pkg/operator/operator.go

```diff
@@ -129,28 +129,36 @@ func (o *Operator) Run(stopc <-chan struct{}) error {
 		return nil
 	}
 
-	go o.worker()
 	go o.cmapInf.Run(stopc)
 
+	glog.V(4).Info("Waiting for initial cache sync.")
+	ok := cache.WaitForCacheSync(stopc, o.cmapInf.HasSynced)
+	if !ok {
+		return errors.New("failed to sync informers")
+	}
+	glog.V(4).Info("Initial cache sync done.")
+
+	go o.worker()
```

Comment:
I think I would also star the worker already as well, just so we're not in a situation where waiting for cache sync wants to enqueue and is blocked on a worker accepting

Reply:
we don't want to start the worker yet, or we will start processing work items and may try to get things out of the cache, which has not finished syncing yet. What will happen here is that the queue will grow until the cache is synced and then the workers will start processing it. The other option is to delay adding the handlers after the sync is finished.

Reply:
the queue does not block adding work items on workers accepting AFAIK.